WordPress/src/main/java/org/wordpress/android/ui/people/WPEditTextWithChipsOutlined.kt

```diff
@@ -0,0 +1,729 @@
+        if (hasText()) {
+            return false
+        }
+
+        // try and remove the last entered username
+        if (flexbox.childCount > 1) {
+            val chipToRemove = flexBox.getChildAt(flexbox.childCount - 2) as Chip
+            val itemName = chipToRemove.text.toString()
+            flexBox.removeViewAt(flexbox.childCount - 2)
+            itemsManager?.onRemoveItem(itemName)
+            return true
+        }
+        return false
+    }
+
+    private fun endsWithDelimiter(string: String): Boolean {
+        if (string.isNullOrBlank()) {
+            return false
+        }
+        for (usernameDelimiter in ITEM_DELIMITERS) {
+            if (string.endsWith(usernameDelimiter)) {
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun styleView(hasFocus: Boolean, hasText: Boolean, animateOnChange: Boolean) {
+        flexBox.background = outlineDrawable
+        setLabelColor(hint, outlineColorDefault, hintLabelColorAlphaDefault)
+        if (hasFocus) {
+            setOutlineStroke(
+                    outlinePixelWidthFocused,
+                    outlineColorFocused,
+                    outlineColorAlphaFocused
+            )
+        } else {
+            setOutlineStroke(
+                    outlinePixelWidthDefault,
+                    outlineColorDefault,
+                    outlineColorAlphaDefault
+            )
+        }
+
+        if (!hasHint()) {
+            hint.visibility = View.INVISIBLE
+            label.visibility = View.INVISIBLE
+            return
+        }
+
+        val previousLabelState = helperTextState
+
+        if (hasFocus || hasText) {
+            helperTextState = HelperTextState.LABEL_VISIBLE
+
+            if (animateOnChange && previousLabelState != helperTextState) {
+                animateViewTo(HelperTextState.LABEL_VISIBLE)
+            } else {
+                if (hasFocus) {
+                    setLabelColor(label, outlineColorFocused, outlineColorAlphaFocused)
+                    hint.visibility = View.INVISIBLE
+                    label.visibility = View.VISIBLE
+                } else {
+                    setLabelColor(label, outlineColorDefault, hintLabelColorAlphaDefault)
+                    hint.visibility = View.INVISIBLE
+                    label.visibility = View.VISIBLE
+                }
+            }
+        } else {
+            helperTextState = HelperTextState.HINT_VISIBLE
+
+            if (animateOnChange && previousLabelState != helperTextState) {
+                animateViewTo(HelperTextState.HINT_VISIBLE)
+            } else {
+                setLabelColor(label, colorSurface, outlineColorAlphaFocused)
+
+                hint.visibility = View.VISIBLE
+                label.visibility = View.INVISIBLE
+            }
+        }
+    }
+
+    private fun animateViewTo(targetState: HelperTextState) {
+        val duration = 167L // this should be the material design default for the animation
```

Comment:
What do you think about having this duration in a global variable of some sort?

Reply:
👍 changed.